nautobot/extras/models/models.py

```diff
@@ -681,3 +685,35 @@ def log(
             else:
                 log_level = logging.INFO
             logger.log(log_level, str(message))
+
+
+@extras_features("graphql")
+class GraphQLQuery(BaseModel, ChangeLoggedModel):
+    name = models.CharField(max_length=100, unique=True)
+    slug = models.CharField(max_length=100, unique=True)
+    query = models.TextField()
+    variables = models.JSONField(encoder=DjangoJSONEncoder, default=dict, blank=True)
+
+    class Meta:
+        ordering = ("slug",)
+
+    def get_absolute_url(self):
+        return reverse("extras:graphqlquery", kwargs={"slug": self.slug})
+
+    def save(self, *args, **kwargs):
+        query_vars = re.findall(r"\$[a-zA-Z0-9]+", self.query)
```

Comment:
Based on feedback from @itdependsnetworks, I wonder if instead of regex whether it would be better to use the same `document_from_string` approach as below to let GraphQL itself parse the query and then retrieve the variable definitions from the parsed document? Something like:

```python
from graphql.language.ast import OperationDefinition
variables = {}
document = backend.document_from_string(schema, self.query)
for definition in document.document_ast.definitions:
    if isinstance(definition, OperationDefinition):
        for variable_definition in definition.variable_definitions:
           default = variable_definition.default_value.value if variable_definition.default_value else ""
           variables[variable_definition.variable.name.value] = default
```

It's obviously more code than a simple regex against the query string, but it's probably "safer" against unanticipated query constructions (for example, what if we have `$` in a string literal somewhere within the query selection set?